tests/api.c

```diff
@@ -34704,7 +34704,163 @@ static void test_wolfSSL_OCSP_get0_info()
+    XFREE(beginTargetBuffer, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    targetBuffer = NULL;
+    ret = wolfSSL_i2d_OCSP_CERTID(&certId, &targetBuffer);
+    // If target buffer is null, function allocates memory for a buffer and
+    // copies the encoded data into it. targetBuffer then points to the start of
+    // this newly allocate buffer.
+    AssertIntEQ(ret, sizeof(rawCertId));
+    for (int i = 0; i < ret; ++i)
+    {
+        AssertIntEQ(targetBuffer[i], rawCertId[i]);
+    }
+
+    XFREE(targetBuffer, NULL, DYNAMIC_TYPE_OPENSSL);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_OCSP_SINGLERESP_get0_id()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP)
+    printf(testingFmt, "wolfSSL_OCSP_SINGLERESP_get0_id()");
+
+    WOLFSSL_OCSP_SINGLERESP single;
+    const WOLFSSL_OCSP_CERTID* certId = wolfSSL_OCSP_SINGLERESP_get0_id(&single);
+    AssertPtrEq(&single, certId);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_OCSP_single_get0_status()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP) 
+    printf(testingFmt, "wolfSSL_OCSP_single_get0_status()");
+
+    WOLFSSL_OCSP_SINGLERESP single;
+    CertStatus certStatus;
+    WOLFSSL_ASN1_TIME* thisDate;
+    WOLFSSL_ASN1_TIME* nextDate;
+    int ret;
+
+    XMEMSET(&single,     0, sizeof(WOLFSSL_OCSP_SINGLERESP));
+    XMEMSET(&certStatus, 0, sizeof(CertStatus));
+
+    // Fill the date fields with some dummy data.
+    for (int i = 0; i < CTC_DATE_SIZE; ++i) {
+        certStatus.thisDateParsed.data[i] = i;
+        certStatus.nextDateParsed.data[i] = i;
+    }
+    certStatus.status = CERT_GOOD;
+    single.status = &certStatus;
+
+    ret = wolfSSL_OCSP_single_get0_status(&single, NULL, NULL, &thisDate,
+                                          &nextDate);
+    AssertIntEQ(ret, CERT_GOOD);
+    AssertPtrEq(thisDate, &certStatus.thisDateParsed);
+    AssertPtrEq(nextDate, &certStatus.nextDateParsed);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_OCSP_resp_count()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP)
+    printf(testingFmt, "wolfSSL_OCSP_resp_count()");
+
+    WOLFSSL_OCSP_BASICRESP basicResp;
+    WOLFSSL_OCSP_SINGLERESP singleRespOne;
+    WOLFSSL_OCSP_SINGLERESP singleRespTwo;
+    int count = 1;
+
+    XMEMSET(&basicResp,     0, sizeof(WOLFSSL_OCSP_BASICRESP));
+    XMEMSET(&singleRespOne, 0, sizeof(WOLFSSL_OCSP_SINGLERESP));
+    XMEMSET(&singleRespTwo, 0, sizeof(WOLFSSL_OCSP_SINGLERESP));
+
+    count = wolfSSL_OCSP_resp_count(&basicResp);
+    AssertIntEQ(count, 0);
+
+    basicResp.single = &singleRespOne;
+    count = wolfSSL_OCSP_resp_count(&basicResp);
+    AssertIntEQ(count, 1);
+
+    singleRespOne.next = &singleRespTwo;
+    count = wolfSSL_OCSP_resp_count(&basicResp);
+    AssertIntEQ(count, 2);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_OCSP_resp_get0()
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_OCSP)
+    printf(testingFmt, "wolfSSL_OCSP_resp_get0()");
```

Comment:
This printf needs to be moved below variable declarations, otherwise some compilers will complain.